lib/puppet/provider/openldap_overlay/olc.rb

```diff
@@ -58,7 +58,12 @@ def self.prefetch(resources)
   end
 
   def exists?
-    @property_hash[:ensure] == :present
+    return @property_hash[:ensure] == :present unless resource[:options]
+
+    # Check overlay options
+    if @property_hash[:options] 
+      (resource[:options].keys - @property_hash[:options].keys).empty? if resource[:options].is_a? Hash
+    end
```

Comment:
Do `options` matching actually define the existence of a given resource? In this case it should be a parameter in the type, not a property.